### PR TITLE
correction in Deprecation messages

### LINKED ISSUE #416: Wrong deprecation message with mergeFeatures
We get wrong warning:

```
library(mia)
data(GlobalPatterns, package="mia")
tse <- GlobalPatterns 
mergeFeatures(tse, f="Genus")
```

> 'mergeRows' is deprecated.
> Use 'mergeFeatures' instead.

-> We should get this deprecation message only when mergeRows is called, not when mergeFeatures is called.

-> To check also for other deprecated functions.

### PR DESCRIPTION
**Pull request by ake123**
Deprecation messages #439
deprecation message about `agglomerateByRank` and `mergeFeaturesByRank`  corrected
Wrong deprecation message with `mergeFeatures` is fixed #416
subset functions and trees #443

### CODE REVIEW

**Comment by artur-sannikov** [comment]
Hey, I'm experiencing this "deprecation message" #439 issue. What's the problem with this pull request?

File: R/agglomerate.R [hunk 1]
```diff
@@ -221,25 +221,26 @@ setMethod("agglomerateByRank", signature = c(x = "SummarizedExperiment"),
...
-setMethod("mergeFeaturesByRank", signature = c(x = "SummarizedExperiment"),
+setMethod("agglomerateByRank", signature = c(x = "SummarizedExperiment"),
           function(x, rank = taxonomyRanks(x)[1], onRankOnly = FALSE, na.rm = FALSE,
                    empty.fields = c(NA, "", " ", "\t", "-", "_"), ...){
-              .Deprecated(old="agglomerateByRank", new="mergeFeaturesByRank", "Now agglomerateByRank is deprecated. Use mergeFeaturesByRank instead.")
-              x <- agglomerateByRank(x, rank = rank, onRankOnly = onRankOnly, na.rm = na.rm,
+              .Deprecated(old="agglomerateByRank", new="mergeFeaturesByRank", msg = "Now agglomerateByRank is deprecated. Use mergeFeaturesByRank instead.")
+              x <- mergeFeaturesByRank(x, rank = rank, onRankOnly = onRankOnly, na.rm = na.rm,
                                      empty.fields = empty.fields, ...)
-              x
```
Comment (TuomasBorman) [inline R/agglomerate.R]: You are not returning anything

File: R/agglomerate.R [hunk 2]
```diff
@@ -257,21 +258,21 @@ setMethod("agglomerateByRank", signature = c(x = "SingleCellExperiment"),
...
-#' @aliases agglomerateByRank
+#' @aliases mergeFeaturesByRank
 #' @importFrom SingleCellExperiment altExp altExp<- altExps<-
 #' @export
-setMethod("mergeFeaturesByRank", signature = c(x = "SingleCellExperiment"),
+setMethod("agglomerateByRank", signature = c(x = "SingleCellExperiment"),
           function(x, ..., altexp = NULL, strip_altexp = TRUE){
-              .Deprecated(old="agglomerateByRank", new="mergeFeaturesByRank", "Now agglomerateByRank is deprecated. Use mergeFeaturesByRank instead.")
-              x <- agglomerateByRank(x, ..., altexp = altexp, strip_altexp = strip_altexp)
-              x
```
Comment (TuomasBorman) [inline R/agglomerate.R]: Same here, the last line is removed

**Comment by TuomasBorman** [comment]
https://github.com/microbiome/mia/pull/404

File: R/agglomerate.R (hunk 2 above)
Comment (ake123) [reply R/agglomerate.R]: fixed

File: R/agglomerate.R (hunk 1 above)
Comment (ake123) [reply R/agglomerate.R]: fixed

**Comment by antagomir** [comment]
Still something ongoing or is this ready?

**Comment by ake123** [comment]
it's ready!

File: R/agglomerate.R [hunk 3]
```diff
@@ -290,26 +293,32 @@ setMethod("agglomerateByRank", signature = c(x = "TreeSummarizedExperiment"),
...
               if(agglomerateTree){
-                  if( length(x@rowTree) > 1 ){
-                      warning("The dataset includes multiple tree after ",
-                              "agglomeration. Agglomeration of tree is not ",
-                              "possible.", call. = FALSE)
-                  } else{
+                  # if( length(x@rowTree) > 1 ){
+                  #     warning("The dataset includes multiple tree after ",
+                  #             "agglomeration. Agglomeration of tree is not ",
+                  #             "possible.", call. = FALSE)
```
Comment (TuomasBorman) [inline R/agglomerate.R]: Why commented?

File: R/merge.R [hunk 4]
```diff
@@ -287,7 +287,7 @@ setMethod("mergeCols", signature = c(x = "SummarizedExperiment"),
 #' @export
 setMethod("mergeFeatures", signature = c(x = "SummarizedExperiment"),
           function(x, f, archetype = 1L, ...){
-              .Deprecated(old="mergeRows", new="mergeFeatures", "Now mergeRows is deprecated. Use mergeFeatures instead.")
+              # .Deprecated(old="mergeRows", new="mergeFeatures", "Now mergeRows is deprecated. Use mergeFeatures instead.")
```
Comment (TuomasBorman) [inline R/merge.R]: Comment

File: R/merge.R [hunk 5]
```diff
@@ -297,7 +297,7 @@ setMethod("mergeFeatures", signature = c(x = "SummarizedExperiment"),
 #' @export
 setMethod("mergeSamples", signature = c(x = "SummarizedExperiment"),
           function(x, f, archetype = 1L, ...){
-              .Deprecated(old="mergeCols", new="mergeSamples", "Now mergeCols is deprecated. Use mergeSamples instead.")
+              # .Deprecated(old="mergeCols", new="mergeSamples", "Now mergeCols is deprecated. Use mergeSamples instead.")
```
Comment (TuomasBorman) [inline R/merge.R]: Just delete, do not leave comments

**Pull request closed without merging**